### ISSUE TITLE
space in favorite or top site tile URL causes Brave to Crash when opening new tab

### ISSUE THREAD

**Issue by LaurenWags** [brave/browser-ios#1597]
### Description:
If a favorite or top site tile on the new tab page has a space in the URL, Brave will not open the new tab page and will crash each time you try to open it.

### Steps to Reproduce

  1. Install 1.6 (18.05.15.13)
  2. Long press on a favorite to get the pencil icon.
  3. Tap on the pencil icon and select 'Edit Favorite'.
  4. Edit the Name and URL. Mine looked like this:
Name: The Verge
URL: https://the Verge.com (space was accidental as my typing of theverge was autocorrected by iOS)
  5. Try to tap on your newly updated icon, the icon gets a bit faded, but nothing happens.
  6. Tap on another icon, the page opens.
  7. Try to open a new tab page.

**Actual result:**
Brave Crashes. If you reopen and try again to open a new tab page, Brave will crash again. 

**Expected result:**
Brave should not crash.

**Reproduces how often:** [Easily reproduced/Intermittent issue]
Easily

**Brave Version:**

1.6 (18.05.15.13)

**Device details:**

**Website problems only:**
- did you check with Brave Shields down?

- did you check in Puffin/Dolphin (UIWebView-based browsers)? 

- did you check in Safari/Firefox (WkWebView-based browsers)?

**Comment by LaurenWags** [comment]
@iccub @jhreis - just to confirm, what I'm seeing after the fix (with 1.6(18.05.16.13)) is if you do manage to get a space in your URL, when you save your changes aren't saved. Just want to confirm if this is expected.

**Comment by iccub** [comment]
@LaurenWags yes, this is expected. Same behavior as editing a regular bookmark.
